samples/excel/09-events/selection-changed.yaml

```diff
@@ -0,0 +1,116 @@
+name: Selection Changed
+description: Add and remove an event handler on the selection changed event
+host: EXCEL
+api_set: {}
+script:
+    content: |
+        $("#add-event-handler").click(addEventHandler);
+        $("#remove-last-event-handler").click(removeLastEventHandler);
+        $("#remove-all-event-handlers").click(removeAllEventHandlers);
+
+        let lastHandler: OfficeExtension.EventHandlerResult<Excel.Workbook>;
+        let workbook: Excel.Workbook;
+
+        async function addEventHandler() {
+            try {
+                if (workbook !== undefined) {
```

Comment:
Just because we have this bug, though, doesn't mean you need to duplicate the code entirely.  Why don't you make the body (the batch) into its own inlined function below, and that wait it's only the Excel.run call that is different?

	async function addEventHandler() {
		try {
			if (workbook === undefined) {
				await Excel.run(addHandlerHelper);
			} else {
				await Excel.run(workbook, addHandlerHelper);
			}
			console.log("Event handler added. Try changing the selection.");
		}
		catch (error) {
			OfficeHelpers.Utilities.log(error);
		}
		

		async function addHandlerHelper(context: Excel.RequestContext) {
			workbook = context.workbook;
			lastHandler = context.workbook.onSelectionChanged.add(onSelectionChanged)

			await context.sync();
		});
	}